catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/RegistryPackageWebConverter.java

```diff
@@ -0,0 +1,1760 @@
+        }
+
+        if (extrinsicObject.isSetMimeType()) {
+            extrinsicObjectMap.put(EXTRINSIC_OBJECT_MIME_TYPE, extrinsicObject.getMimeType());
+        }
+
+        if (extrinsicObject.isSetExternalIdentifier()) {
+            putExternalIdentifier(extrinsicObject.getExternalIdentifier(), extrinsicObjectMap);
+        }
+
+        if (!extrinsicObjectMap.isEmpty()) {
+            registryObjectListMap.putIfAbsent(EXTRINSIC_OBJECT_KEY,
+                    new ArrayList<Map<String, Object>>());
+            ((List) registryObjectListMap.get(EXTRINSIC_OBJECT_KEY)).add(extrinsicObjectMap);
+        }
+    }
+
+    private static void populateROLWithExtrinsicObject(
+            List<Map<String, Object>> extrinsicObjectsMapList,
+            RegistryObjectListType registryObjectList) {
+        for (Map<String, Object> extrinsicMap : extrinsicObjectsMapList) {
+            ExtrinsicObjectType extrinsicObject = getExtrinsicObjectFromMap(extrinsicMap);
+
+            if (extrinsicObject != null) {
+                registryObjectList.getIdentifiable()
+                        .add(RIM_FACTORY.createExtrinsicObject(extrinsicObject));
+            }
+        }
+    }
+
+    private static void populateROLWithService(List<Map<String, Object>> serviceMapList,
+            RegistryObjectListType registryObjectList) {
+        for (Map<String, Object> serviceMap : serviceMapList) {
+            ServiceType service = getServiceFromMap(serviceMap);
+
+            if (service != null) {
+                registryObjectList.getIdentifiable()
+                        .add(RIM_FACTORY.createService(service));
+            }
+        }
+    }
+
+    private static void populateROLWithOrganization(List<Map<String, Object>> organizationMapList,
+            RegistryObjectListType registryObjectList) {
+        for (Map<String, Object> organizationMap : organizationMapList) {
+            OrganizationType organization = getOrganizationFromMap(organizationMap);
+
+            if (organization != null) {
+                registryObjectList.getIdentifiable()
+                        .add(RIM_FACTORY.createOrganization(organization));
+            }
+        }
+    }
+
+    private static void populateROLWithPerson(List<Map<String, Object>> mapList,
+            RegistryObjectListType registryObjectList) {
+        for (Map<String, Object> personMap : mapList) {
+            PersonType person = getPersonFromMap(personMap);
+
+            if (person != null) {
+                registryObjectList.getIdentifiable()
+                        .add(RIM_FACTORY.createPerson(person));
+            }
+        }
+    }
+
+    private static void populateROLWithAssociation(List<Map<String, Object>> mapList,
+            RegistryObjectListType registryObjectList) {
+        for (Map<String, Object> associationMap : mapList) {
+            AssociationType1 association = getAssociationFromMap(associationMap);
+
+            if (association != null) {
```

Comment:
Instead of using all those `null` checks and `populated` flags, you may want to consider using Java's `Optional<>`. Should make your code cleaner and reduce complexity quite a bit.